crates/ethereum-instance/src/lib.rs

```diff
@@ -47,45 +41,46 @@ pub enum Error {
     ContractBuildArtifacts(&'static str),
 }
 
-/// A convenient wrapper over an already running or spawned Anvil local devnet
-pub struct EthereumSandbox {
+/// A convenient wrapper over an already running or spawned Anvil local devnet or ethereum 
+pub struct EthereumClient {
     /// If initialized keeps an Anvil instance to properly shutdown it at the end
-    _anvil: Option<AnvilInstance>,
+    client: Option<AnvilInstance>,
     /// Pre-configured local signer
-    client: Arc<LocalWalletSignerMiddleware>,
+    signer: Arc<LocalWalletSignerMiddleware>,
 }
 
-impl EthereumSandbox {
+impl EthereumClient {
     /// Creates a new sandbox instance.
-    /// Will try to attach to already running Anvil instance using one
-    /// of the following endpoints:
-    ///     - `anvil_endpoint` parameter (if specified)
-    ///     - ${ANVIL_ENDPOINT} environment variable (if set)
-    ///     - http://127.0.0.1:8545 (default)
-    /// Also default values for chain ID and private keys will be used.
-    pub fn attach(anvil_endpoint: Option<String>) -> Result<Self, Error> {
-        let anvil_endpoint = anvil_endpoint.unwrap_or_else(|| {
+    /// Will try to attach to already running Anvil instance or custom rpc and private key provided to the function.
+    /// if not provided any argument it will attack to a default anvil instance with default anvil params.
+    pub fn attach(rpc_endpoint: Option<String>, priv_key: Option<String>) -> Result<Self, Error> {
+        let rpc_endpoint = rpc_endpoint.unwrap_or_else(|| {
             std::env::var("ANVIL_ENDPOINT")
```

Comment:
let's rename this as well to make it generic, `ETH_RPC_ENDPOINT`? we will also need to change it in Madara